mesop/components/html/html.py

```diff
@@ -6,23 +8,31 @@
   insert_component,
   register_native_component,
 )
+from mesop.warn import warn
 
 
 @register_native_component
 def html(
   html: str = "",
   *,
+  mode: Literal["sanitized", "sandboxed"] = "sanitized",
   style: Style | None = None,
   key: str | None = None,
 ):
   """
-  This function renders custom HTML inside an iframe for web security isolation.
+  This function renders custom HTML in a secure way.
 
   Args:
       html: The HTML content to be rendered.
+      mode: The mode of the iframe. Can be either "sanitized" or "sandboxed". If "sanitized" then potentially dangerous content like `<script>` and `<style>` are
+          stripped out. If "sandboxed", then all content is allowed, but rendered in an iframe for isolation.
       style: The style to apply to the embed, such as width and height.
       key: The component [key](../components/index.md#component-key).
   """
+  if mode != "sandboxed" and ("<script>" in html or "<style>" in html):
```

Comment:
@richard-to WDYT about having a default mode (where mode is not specified) where it uses sandboxed if it detects a <script> or <style> and then sanitized otherwise? I think this would produce the right behavior in _most_ circumstances, although it's definitely possible it'll over-trigger for sandboxed (which isn't the worst thing). Or do you think that is too magical?

Reply:
Yeah that's a good question. Personally I like the more explicit modes here since it requires the developer to make the choice for what they intend. So hopefully they would know the pros/cons of their choice.

But I also agree that there is the edge case where someone has like some text that says "<script>" somewhere that could cause the sandbox to trigger. Seems rare and could be preventable with a more robust check for script and style tags.

In the end, I don't feel strongly one way or the other here. Both seem fine, especially if the iframe can be rendered to the correct height automatically.

Reply:
OK, I've done the implicit mode since I've automatically resized the iframe.